File: teccon-system-fe/src/interfaces/CapsuleResult.ts
import { PressType } from "./enums/PressType";

export interface CapsuleResult {
    id: number;
    sampleId: number;
    curingAgeDays: number;
    failureLoadKgf: number | null;      
    compressiveStrengthMpa: number | null; 
    pressType: PressType | null;
    testedAt: string | null;             
}

File: teccon-system-be/teccon/src/main/java/com/example/teccon/testPhase/CapsuleResult.java
package com.example.teccon.testPhase;

import com.example.teccon.collectionPhase.Sample;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "capsule_results")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CapsuleResult {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "sample_id",
            foreignKey = @ForeignKey(name = "fk_capsule_sample"))
    private Sample sample;

    private Integer curingAgeDays;
    private Double failureLoadKgf;
    private Double compressiveStrengthMpa;

    @Column(length = 50)
    private String pressType;

    private LocalDateTime testedAt;
}